x-pack/plugins/endpoint/server/plugin.test.ts

```diff
@@ -3,10 +3,44 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+/* eslint-disable max-classes-per-file */
 
+import { SavedObjectsClientContract } from 'kibana/server';
 import { EndpointPlugin, EndpointPluginSetupDependencies } from './plugin';
 import { coreMock } from '../../../../src/core/server/mocks';
 import { PluginSetupContract } from '../../features/server';
+import { IndexPatternService } from '../../ingest_manager/server';
+
+export const MetadataIndexPattern = 'metadata-endpoint-*';
+
+export interface IndexPatternRetriever {
+  get(client: SavedObjectsClientContract, datasetPath: string, version?: string): Promise<string>;
```

Comment:
get may be too generic and probably assumes only one method in this interface. It also may not covey much.

Reply:
Yeah that's fair, how about `getIndexPattern` instead?

Reply:
Yes that will say what it is doing